src/field/captcha/third_party_captcha.jsx

```diff
@@ -84,39 +119,88 @@ export class ThirdPartyCaptcha extends React.Component {
     };
   }
 
-  static loadScript(props, element = document.body, callback = noop) {
-    const callbackName = `${providerDomPrefix(props.provider)}Callback_${Math.floor(Math.random() * 1000001)}`;
-    const scriptUrl = scriptForProvider(props.provider, props.hl, callbackName);
-    const script = document.createElement('script');
-
-    window[callbackName] = () => {
-      delete window[callbackName];
-      callback(null, script);
+  injectCaptchaScript(callback = noop) {
+    const { provider, hl, clientSubdomain, sitekey } = this.props;
+    const callbackName = `${providerDomPrefix(provider)}Callback_${Math.floor(Math.random() * 1000001)}`;
+    const scriptUrl = scriptForProvider(provider, hl, callbackName, clientSubdomain, sitekey);
+    const attributes = {
+      async: true,
+      defer: true
     };
-
-    script.src = scriptUrl;
-    script.async = true;
-    script.defer = true;
-    if (props.provider === FRIENDLY_CAPTCHA_PROVIDER) {
-      script.onload = window[callbackName];
+    if (provider === ARKOSE_PROVIDER) {
+      attributes['data-callback'] = callbackName;
+      attributes['onerror'] = function () {
+        if (this.state.retryCount < MAX_RETRY) {
+          removeScript(scriptUrl);
+          loadScript(scriptUrl, attributes);
+          this.setState((prevState) => ({
+            retryCount: prevState.retryCount + 1
+          }));
+          return;
+        }
+        removeScript(scriptUrl);
+        this.changeHandler('BYPASS_CAPTCHA');
+      };
+      window[callbackName] = function (arkose) {
+        callback(arkose);
+      };
+    } else {
+      window[callbackName] = () => {
+        delete window[callbackName];
+        callback();
+      };
+  
+      if (provider === FRIENDLY_CAPTCHA_PROVIDER) {
+        attributes['onload'] = window[callbackName];
+      }
     }
 
-    element.appendChild(script);
+    loadScript(scriptUrl, attributes);
   }
 
   componentWillUnmount() {
-    if (!this.scriptNode) {
+    if (!this.state.scriptUrl) {
       return;
     }
-    document.body.removeChild(this.scriptNode);
+    removeScript(this.state.scriptUrl);
   }
 
   componentDidMount() {
-    ThirdPartyCaptcha.loadScript(this.props, document.body, (err, scriptNode) => {
-      this.scriptNode = scriptNode;
+    this.injectCaptchaScript((arkose) => {
       const provider = getCaptchaProvider(this.props.provider);
-
-      if (this.props.provider === FRIENDLY_CAPTCHA_PROVIDER) {
+      if (this.props.provider === ARKOSE_PROVIDER) {
+        var handleArkose = function (event) {
+          setTimeout(function () {
+            arkose.run();
+          }, TIMEOUT_MS);
```

Comment:
Is there some information on why we need to use a setTimeout here?

Reply:
No will be removed.